mountpoint-s3-client/tests/endpoint.rs

```diff
@@ -62,29 +63,41 @@ async fn test_addressing_style_uri(addressing_style: AddressingStyle) {
 #[tokio::test]
 async fn test_addressing_style_uri_dualstack(addressing_style: AddressingStyle) {
     run_test(|region| {
-        let uri = format!("https://s3.dualstack.{region}.amazonaws.com");
+        let domain = get_test_domain(region);
+        let uri = format!("https://s3.dualstack.{region}.amazonaws.{domain}");
         Endpoint::from_uri(&uri, addressing_style).unwrap()
     })
     .await;
 }
 
 // FIPS endpoints can only be used with virtual-hosted-style addressing
+#[cfg(feature = "fips_tests")]
 #[test_case(AddressingStyle::Virtual)]
 #[tokio::test]
 async fn test_addressing_style_uri_fips(addressing_style: AddressingStyle) {
     run_test(|region| {
-        let uri = format!("https://s3-fips.{region}.amazonaws.com");
+        let domain = get_test_domain(region);
+        let uri = format!("https://s3-fips.{region}.amazonaws.{domain}");
         Endpoint::from_uri(&uri, addressing_style).unwrap()
     })
     .await;
 }
 // FIPS endpoints can only be used with virtual-hosted-style addressing
+#[cfg(feature = "fips_tests")]
 #[test_case(AddressingStyle::Virtual)]
 #[tokio::test]
 async fn test_addressing_style_uri_fips_dualstack(addressing_style: AddressingStyle) {
     run_test(|region| {
-        let uri = format!("https://s3-fips.dualstack.{region}.amazonaws.com");
+        let domain = get_test_domain(region);
+        let uri = format!("https://s3-fips.dualstack.{region}.amazonaws.{domain}");
         Endpoint::from_uri(&uri, addressing_style).unwrap()
     })
     .await;
 }
+
+fn get_test_domain(region: &str) -> String {
+    match region {
+        "cn-north-1" | "cn-northwest-1" => "com.cn".to_string(),
+        _ => "com".to_string(),
+    }
+}
```

Comment:
Can we do this as an environment variable?

Other partitions may have different suffixes again, not just the `aws-cn` partition.

Something like:

```
AWS_REGION_DOMAIN_SUFFIX=amazonaws.com
AWS_REGION_DOMAIN_SUFFIX=amazonaws.com.cn
```

Of course, default is `amazonaws.com` if not set.

Reply:
I don't think this function should exist at all -- all these tests need to be rewritten to use the endpoint resolver once it gains support for setting the FIPS/dualstack/etc flags. At that point none of these tests should have any hardcoded URI components at all, they should just be different `EndpointConfig`s or whatever we end up calling the way you configure the endpoint resolver.

Reply:
Sorry, just saw this. You're right.

Not sure when that'll be in, but hopefully this is something we can remove very soon.